### ISSUE TITLE
Nginx with mod_pagespeed integration on a https website.

### ISSUE THREAD

**Issue by ghost** [apache/incubator-pagespeed-mod#1694]
Trying to setup pagespeed for a https website, the assets are being rejected 'Cannot fetch url 'https://www.domain.com/index.php': as https is not supported'.

I've checked out https://www.modpagespeed.com/doc/https_support#https_fetch
and I've added below to my pagespeed.conf
`pagespeed FetchHttps enable,allow_self_signed;`

and inside my domain.conf

```
pagespeed Domain https://www.domain.com;
pagespeed SslCertDirectory /etc/ssl/certs;
pagespeed SslCertFile /etc/ssl/certs/domain.privkey.crt;
```
I'm not sure where to point the SslCertFile? Its not my domain ssl file, is it my csr or the private key?

Is there anything else that I need to add to my domain.conf file?

Env info
Running Nginx 1.13.3 with pagespeed 1.12.34.2 inside a docker instance on an AWS ec2 instance with no loadbalancer.

**Comment by oschaaf** [comment]
Did you see https://www.modpagespeed.com/doc/https_support#configuring_ssl_certificates ?

**Comment by ghost** [comment]
Thanks for the reply.

Yes, I have looked at that, and I have tried an array of different files for SslCertFile, such as

- `/etc/ssl/certs/COMODO_RSA_Certification_Authority.crt`
- `/etc/ssl/certs/AddTrust_External_CA_Root.crt`
- `/etc/ssl/certs/comodo.ca-bundle`
- and desperately tried both domain CSR `/etc/ssl/private/domain.key` and SSL `/etc/ssl/cert/domain.crt`.

It still coming up with the same warning "https is no supported".

Am i over looking something?

**Comment by oschaaf** [comment]
Are you using the native fetcher feature in ngx_pagespeed? Could you paste the full pagespeed configuration?

**Comment by ghost** [comment]
Yeah, the native fetcher has always been on and still not processing the cert.
I've included most/all of my config [here](https://github.com/AUSWinner/nginx-pagespeed-redis/tree/master/conf)

- [domain conf](https://github.com/AUSWinner/nginx-pagespeed-redis/blob/master/conf/conf.d/domain.com.conf), and 
- [pagespeed conf](https://github.com/AUSWinner/nginx-pagespeed-redis/blob/master/conf/conf.d/pagespeed.conf)

I appreciate all the help thus far.

**Comment by oschaaf** [comment]
The native fetcher does not support https. Unfortunatly it looks like the docs do not mention that. Turning it off should help your problem.

**Comment by ghost** [comment]
Providing an update after playing around with the config settings.

It appears that the `COMODO_RSA_Certification_Authority.crt` isn't the correct cert, it looks like the most correct cert to use is the `AddTrust_External_CA_Root.crt`, but it didn't work until I've added `allow_unknown_certificate_authority` to the `FetchHttps`. 

Is there a list of verified CA? Is there any draw backs in using an unverified CA?
